googledatastore/src/main/java/com/yahoo/ycsb/db/GoogleDatastoreClient.java

```diff
@@ -0,0 +1,334 @@
+
+  private String rootEntityName;
+
+  private Datastore datastore = null;
+
+  public GoogleDatastoreClient() {}
+
+  /**
+   * Initialize any state for this DB. Called once per DB instance; there is
+   * one DB instance per client thread.
+   */
+  @Override
+  public void init() throws DBException {
+    String debug = getProperties().getProperty("googledatastore.debug", null);
+    if (null != debug && "true".equalsIgnoreCase(debug)) {
+        logger.setLevel(Level.DEBUG);
+    }
+
+    // We need the following 3 essential properties to initialize datastore:
+    //
+    // - DatasetId,
+    // - Path to private key file,
+    // - Service account email address.
+    String datasetId = getProperties().getProperty(
+        "googledatastore.datasetId", null);
+    if (datasetId == null) {
+      throw new DBException(
+          "Required property \"datasetId\" missing.");
+    }
+
+    String privateKeyFile = getProperties().getProperty(
+        "googledatastore.privateKeyFile", null);
+    if (privateKeyFile == null) {
+      throw new DBException(
+          "Required property \"privateKeyFile\" missing.");
+    }
+
+    String serviceAccount = getProperties().getProperty(
+        "googledatastore.serviceAccount", null);
+    if (serviceAccount == null) {
+      throw new DBException(
+          "Required property \"serviceAccount\" missing.");
+    }
+
+    // Below are properties related to benchmarking.
+
+    String readConsistencyConfig = getProperties().getProperty(
+        "googledatastore.readConsistency", null);
+    if (readConsistencyConfig != null) {
+      try {
+        this.readConsistency = ReadConsistency.valueOf(
+            readConsistencyConfig.trim().toUpperCase());
+      } catch (IllegalArgumentException e) {
+        throw new DBException("Invalid read consistency specified: " +
+            readConsistencyConfig + ". Expecting STRONG or EVENTUAL.");
+      }
+    }
+
+    //
+    // Entity Grouping Mode (googledatastore.entitygroupingmode), see
+    // documentation in conf/googledatastore.properties.
+    //
+    String entityGroupingConfig = getProperties().getProperty(
+        "googledatastore.entityGroupingMode", null);
+    if (entityGroupingConfig != null) {
+      try {
+        this.entityGroupingMode = EntityGroupingMode.valueOf(
+            entityGroupingConfig.trim().toUpperCase());
+      } catch (IllegalArgumentException e) {
+        throw new DBException("Invalid entity grouping mode specified: " +
+            entityGroupingConfig + ". Expecting ONE_ENTITY_PER_GROUP or " +
+            "MULTI_ENTITY_PER_GROUP.");
+      }
+    }
+
+    this.rootEntityName = getProperties().getProperty(
+        "googledatastore.rootEntityName", "YCSB_ROOT_ENTITY");
+
+    try {
+      // Setup the connection to Google Cloud Datastore with the credentials
+      // obtained from the configure.
+      DatastoreOptions.Builder options = new DatastoreOptions.Builder();
```

Comment:
@stfeng can we somehow support both ADC and credential file authentication in there? Like adding a flag to switch between these 2 modes. This will be very handy.